### ISSUE TITLE
magit-blame with hunk movement

### ISSUE THREAD

**Issue by kvaneesh** [magit/magit#1777]
Is there an easy way to follow code that got moved across files using magit blame (recursive). Here is what i am trying todo.
1) start with magit-blame on a buffer. I am interested in hunk A
2) Found that hunk A got moved from file B to file A.
3) Now how do i continure blame of file B ?

One possible way would be add magit-show-this-file <commit> which shows the content of that  file as per commit. Then if we can allow recursive blame of the buffer, I could get the job done. For my particular use case, I would get the commit from file A blame buffer, I use magit-show-commit which takes me to diff which has both the file A and file B. Now, we could do C-u RET on removed hunk of file B. That can be made to show content of file B with commit-line ~1 ?. Then i could recursive blame that buffer ?

I know this is a feature request, But not sure how to tag this as one.

**Comment by tarsius** [comment]
That already works on the `next` branch, pretty much as you described. There was a small issue which causes point to move to the beginning of the buffer when blaming a buffer which visits `rev:file`. I have fixed that now, so:

2) Found that hunk A got moved from file B to file A.

I assume you did so by pressing `RET` while in the blamed file. This shows a diff. Then you moved to the hunk that "adds" the piece of code to file B.

3) Now how do i continure blame of file B ?

Press `RET` you are now in a buffer visiting B as it was in the commit that added the lines you are interested in. Do `M-x magit-blame` again. Rinse and repeat.

**Comment by kvaneesh** [comment]
Ok I got one more problem related to this. When i select B it will show the version of file B at that commit  (say 'C') . ie, there are no new line in the file B related to that commit C, (no new lines added to B). All that the commit did was to move the code from B to A. Now how do i get the version of file B before that commit C ? . For recursive blame to work, I need a line in file B which got added as part of C right ?

-aneesh
